src/compas_viewer/scene/geometryobject.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional
+from typing import Tuple
+
+from compas.colors import Color
+from compas.geometry import Geometry
+from compas.geometry import Line
+from compas.geometry import Point
+from compas.scene import GeometryObject as BaseGeometryObject
+
+from .sceneobject import ShaderDataType
+from .sceneobject import ViewerSceneObject
+
+
+class GeometryObject(ViewerSceneObject, BaseGeometryObject):
```

Comment:
it might make sense to differentiate geometry objects further (for example into curves, surfaces, and shapes), because this might make conversion of parametric geometry into discrete geometry more efficient, but we can do that at a later stage...